cmd/util/ledger/migrations/cadence.go

```diff
@@ -41,6 +41,28 @@ func NewInterfaceTypeConversionRules(chainID flow.ChainID) StaticTypeMigrationRu
 	}
 }
 
+var typeRequirementRemovals = []struct {
```

Comment:
Nice! Maybe we can generalize this and infer it from an update (analyze old -> new)? 

Basically all nested composite types that were declared inside of a contract interface can be assumed to have to be changed to interface types.

Reply:
yeah, can do that. Going to need another migration step (doesn't migrate anything), to extract this from the existing payloads. That needs to be run before the staged contract migration.

Reply:
added in the recent commits